students/Cracolici_Jon/lesson05/JonCracolici_Mailroom3.py

```diff
@@ -0,0 +1,180 @@
+    elif updonordict['donornamekey'] not in dB.keys():
+        print("It appears that you are attempting t update a donor not already in our system.\n Perhaps you would like to select the 'New Donor' option instead?")
+        return False
+    #will add try except to check for donor actually being new in L5
+    update_donor(updonordict, dB)
+    note_gen(dB[updonordict['donornamekey']])
+    return dB#, start_control(dB)
+def donor_input(dB):
+    """This function collects user input about the donation."""
+    #create a set of quit criteria
+    quitset = set(['quit', 'q'])
+    donorname = input('Who is the donor?')
+    #Check if user wants to exit to previous menu
+    if donorname.lower() in quitset:
+        return False
+    newdonation = input('How much did they donate?')
+    #Check if user wants to exit to previous menu
+    if newdonation.lower() in quitset:
+        return False
+        
+    else:#this section creates the keys and values for the outputted dict
+        donornamekey = donorname.lower()
+        donornamekey = donornamekey.replace(' ', '_')
+        donor = donorname.title()
+        try:
+            donation = float(newdonation)
+        except ValueError:
+            print("Use numbers for the donation amount. NUMBERS!")
+            return False
+        newdondict = {'donornamekey': donornamekey, 'donor': donor, 'donation': donation}
+        return newdondict
+def print_letters(donordict):
+    """This function takes a string and makes a document with it."""
+    lettername = donordict['fname']+'.txt'
+    outfile = open(lettername, 'w')
+    outfile.write(note_gen(donordict, dest = 'f'))
+    outfile.close()
+def note_gen(persondict, dest = 's'):
+    """Creates a thank you note. Includes a required positional argument of the intended recipient's info,
+    and a kwarg "dest" that defaults to printing to screen but may be set to generate a file."""
+    message = "Dear {donor}, \nThank you for your generous donation of ${lastdonation:.02f}. Please rest assured that we will use at least \n95% of your contribution to feed the homeless to wolves. We could not do this work without you. \nSincerely, \nThe Billionaires' Club".format(**persondict)
+    if dest == 's':
+        print(message)
+    elif dest == 'f':
+        return message
+def quit_program(db):
+    print("Goodbye dear User!")
+    return False
+    #sys.exit()
+def quit_saty(db):
+    print("Bye bye!")
+    return False
+def main():
+    db = initialize_database()
+    while True:
+        x = start_control(db)
+        if x == False:
+            break
+def main_2(db):
+    while True:
+        x = saty_control(db)
+        if x == False:
+            break
+#### This is the beginning of the executed program.
+if __name__=="__main__":
+
+    startmenulogic = {'1': main_2, '2': create_report, '3': mass_mail, '4': quit_program}
+    satymenulogic = {'1': show_current_names, '2': ud_control, '3': nd_control, '4': quit_saty}
+    main()
```

Comment:
recommend simplifying your main entry point, right now there are a lot of layers when it can be more straightforward. Example:
```
def quit_program():
    print("Exiting")
    sys.exit()


def main():
    while True:
        response = input("")
        try:
            switch_dict[response]()
        except KeyError:
            print("Invalid option")

....
switch_dict = {'1': send_thank_you, '2': create_report, '3': mass_mail, '4': quit_program}
main()
```

this is all you really need to do:
* simple while True loop that should only break on "quit" option
* quit is  mapped to sys.exit which simply exits your script
* no special return statement in feature functions to control main flow when it shouldn't really